### ISSUE TITLE
Questions on item embedding computation

### ISSUE THREAD

**Issue by cavacH** [bytedance/HLLM#35]
Hi, thanks for the great work! I have several questions on the item embedding extraction process:
1. If I'm understanding correctly, item tokens are concatenated into one unified sequence for one user sequence (one training datapoint): https://github.com/bytedance/HLLM/blob/main/code/REC/data/dataset/trainset.py#L207-L217, and the item embedding is designated as the hidden state at the special token's position: https://github.com/bytedance/HLLM/blob/main/code/REC/model/HLLM/hllm.py#L164
It's a little counter-intuitive to me that all item tokens are inferred within one large sequence. Will the context between items influence each other in this way? I originally thought the item embeddings are inferred with batch, and independently.

2. I noticed that there are optimizations based on flash attention, implemented by additional `cu_input_lens` argument during inference. So does the codebase guarantee the same behavior while running on GPUs not supporting flash attention (like V100)? Just want to make sure if I could possibly do some small experiments on V100 GPUs by disabling flash attention (`use_ft_flash_attn` set to false).

**Comment by ssyzeChen** [comment]
Hi~
1. flash attn op ensures that different items will not influence each other during forward.
2. You could turn off flash attn by https://github.com/bytedance/HLLM/blob/main/code/REC/model/HLLM/hllm.py#L42, but we are sorry for not supporting this yet, you'll need to modify the training and serving datasets to adapt to the original attention implementation. This involves removing cu_input_lens, inputting input_ids and attention_mask, and using the correct mask.
